src/serac/infrastructure/debug_print.hpp

```diff
@@ -74,7 +74,11 @@ std::ostream& operator<<(std::ostream& out, serac::SignedIndex i)
  * @param filename the name of the output file
  */
 template <typename T>
+#ifdef SERAC_USE_UMPIRE
 void write_to_file(axom::Array<T, 2, axom::MemorySpace::Host> arr, std::string filename)
+#else
```

Comment:
You would think that `Host` would always be there but its guarded in axom and `Dynamic` is the one unguarded....

https://github.com/LLNL/axom/blob/2244835b3389c427c337b5df782982d2ae609bfe/src/axom/core/memory_management.hpp#L37-L47

Reply:
Seems like host should be safe to unguard in addition to dynamic?  Regardless, we could use dynamic + umpire allocator IDs to achieve the same without the clunky #ifdef blocks.  Though we would lose the explicit typing of the memory.  Thoughts?

Reply:
I say go with `Dynamic` for now to avoid the ifdef's...

Reply:
@white238 and I spoke -- it makes sense to go with (the default) Dynamic

Reply:
It's not clear to me Dynamic will give the desired result for the case where the original `axom::Array` is on device.  It seems like this copy constructor will be called: https://github.com/LLNL/axom/blob/develop/src/axom/core/Array.hpp#L1180-L1190 which copies the allocator ID from the existing array.  The result will be an array on device and the operator() call on L88 will fail.  Is there a different copy constructor being called by this function?

Reply:
Implemented a fix suggested by @kennyweiss